activesupport/lib/active_support/testing/setup_all_and_teardown_all.rb

```diff
@@ -0,0 +1,56 @@
+require "active_support/concern"
+require "active_support/callbacks"
+
+module ActiveSupport
+  module Testing
+    # Add support for +setup_all+ and +teardown_all+ callbacks.
+    module SetupAllAndTeardownAll
+      extend ActiveSupport::Concern
+
+      included do
+        include ActiveSupport::Callbacks
+        define_callbacks :setup_all, :teardown_all
+      end
+
+      module ClassMethods
+        def run(reporter, options = {})
+          @reporter = reporter
+          @instance = run_callbacks :setup_all
+          super(reporter, options)
+          run_callbacks :teardown_all
+        end
+
+        # Return a singleton instance for running individual tests,
+        # so instance variables can be duplicated across all tests and callbacks.
+        def new(name)
+          @instance ||= super
+          instance = @instance.dup
+          instance.name = name
+          instance.failures = []
+          instance
+        end
+
+        private
+
+          def run_callbacks(name)
+            instance = new("run_callbacks")
```

Comment:
Due to some internal test-reporter implementation, the name here needs to be the name of an available method. I don't know if `"run_callbacks"` is too meaningful, but this prevents any errors when there's an error raised in one of the callbacks